### PR TITLE
Issue #1045 ghb allocation

### LINKED ISSUE #1045: Create converter for iMOD5 GHB packages
It needs to:

- Allocate planar grids to layers

- Distribute conductances of planar grids. 
It should be fairly  similar to the import of the drainage package, but it was converted into a separate issue because there is no imod5 test-data containing ghb boundaries. So that should be obtained first when starting on this issue.

### PR DESCRIPTION
**Pull request by luitjansl**
Fixes #1045

# Description
Enables using allocation options when importing a ghb from imod5 and it is planar (layer=0)

# Checklist

- [X] Links to correct issue
- [ ] Update changelog, if changes affect users
- [X] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [X] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/mf6/ghb.py [hunk 1]
```diff
@@ -207,19 +220,54 @@ def from_imod5_data(
...
         regridded_package_data = _regrid_package_data(
             data, idomain, regridder_types, regrid_cache, {}
         )
+        if is_planar:
+            conductance = regridded_package_data["conductance"]
+
+            planar_head = regridded_package_data["head"]
+            k = target_npf.dataset["k"]
+
+            ghb_alocation = allocate_ghb_cells(
```
Comment (JoerivanEngelen) [inline imod/mf6/ghb.py:246]: Typo:

```suggestion
            ghb_allocation = allocate_ghb_cells(
```

File: imod/mf6/ghb.py [hunk 2]
```diff
@@ -207,19 +220,54 @@ def from_imod5_data(
...
+            )
+
+            layered_head = planar_head.where(ghb_alocation)
+            layered_head = enforce_dim_order(layered_head)
+
+            regridded_package_data["head"] = layered_head
+
+            if "layer" in conductance.coords:
+                conductance = conductance.isel({"layer": 0})
+                conductance = conductance.drop_vars("layer")
```
Comment (JoerivanEngelen) [inline imod/mf6/ghb.py:261]: Tip, you can shorten this to:

```suggestion
                conductance = conductance.isel({"layer": 0}, drop=True)
```

**Review by JoerivanEngelen — APPROVED** [review]
Looks good, thanks for also fixing the docstrings in other packages. I have two minor comments. Approving in advance.

File: imod/mf6/ghb.py (hunk 1 above)
Comment (luitjansl) [reply imod/mf6/ghb.py]: done
